node/trieIterators/factory/sovereignDirectStakedListHandlerFactory_test.go

```diff
@@ -0,0 +1,40 @@
+package factory_test
+
+import (
+	"fmt"
+	"github.com/multiversx/mx-chain-go/node/trieIterators"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/node/mock"
+	trieIteratorsFactory "github.com/multiversx/mx-chain-go/node/trieIterators/factory"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	stateMock "github.com/multiversx/mx-chain-go/testscommon/state"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSovereignDirectStakedListProcessorFactory(t *testing.T) {
+	t.Parallel()
+
+	directStakedListHandlerFactory := trieIteratorsFactory.NewSovereignDirectStakedListProcessorFactory()
+	require.False(t, directStakedListHandlerFactory.IsInterfaceNil())
+}
+
+func TestSovereignDirectStakedListHandlerFactory_CreateDirectStakedListProcessorHandler_DirectStakedListHandlerFactory(t *testing.T) {
+	t.Parallel()
+
+	args := trieIterators.ArgTrieIteratorProcessor{
```

Comment:
We could've created a func for this duplicated code to create the args that can receive the shardID as paramter, or change it afterwards